samples/java-spring-transfer-workflow/src/main/java/com/example/wallet/WalletEntity.java

```diff
@@ -36,11 +33,10 @@ public Effect<String> create(@PathVariable String id, @PathVariable int initBala
 
   @PatchMapping("/withdraw/{amount}") // <2>
   public Effect<String> withdraw(@PathVariable int amount) {
-    var newBalance = currentState().balance() - amount;
-    if (newBalance < 0) {
+    Wallet updatedWallet = currentState().withdraw(amount);
```

Comment:
Yeah, why not, just please add similar changes to `java-spring-transfer-workflow-compensation`